pages/desktop/user.py

```diff
@@ -119,8 +120,19 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        WebDriverWait(self.selenium, self.timeout).until(expected_conditions.visibility_of(email_checkbox));
+        if not email_checkbox.is_selected():
+                self.selenium.find_element(*self._email_checkbox).click()
```

Comment:
You have a good point about the extra wait I suggested, and the fact that the state isn't likely to change over time, so an `if` is appropriate, but I think you need to fail the test if the `if` evaluates to indicate that the expected state is not present. For example, in this current code, if the checkbox is currently selected, rather than fail the test you simply do not click on the checkbox. I suggest that is the checkbox is currently selected, but we expected it not to be selected, then we need to fail the test.

The `return email_checkbox.is_selected() == True` that you add below doesn't actually do anything, as you are not checking the return value from this method (nor would I expect you to). I am guessing that this `return` was an attempt to notify the test if the page is not in the expected state after clicking the checkbox, but it will not do that.

What would do that would be to replace this `return` with a wait that does wait for the checkbox to be in the correct state. If you make those suggested changes then the following should be in place:

1. If the checkbox is not in the expected state before clicking - fail the test.
2. If the checkbox is not in the expected after before clicking - fail the test (via a wait timeout).